tests/unit/core/components/TestServerHtml.js

```diff
@@ -39,135 +35,136 @@ describe(__filename, () => {
+    const root = render();
+    const js = root.find('script');
+
+    expect(js.at(1)).toHaveProp('src', '/foo/disco-blah.js');
   });
 
   it('renders css with SRI when present', () => {
-    const html = findRenderedDOMComponentWithTag(render(), 'html');
-    const styleSheet = html.querySelector('link[rel="stylesheet"]');
-    expect(styleSheet.getAttribute('integrity')).toEqual('sha512-disco-css');
-    expect(styleSheet.getAttribute('crossorigin')).toEqual('anonymous');
+    const root = render();
+    const styleSheets = root.find({ rel: 'stylesheet' });
+
+    expect(styleSheets.at(0)).toHaveProp('integrity', 'sha512-disco-css');
+    expect(styleSheets.at(0)).toHaveProp('crossOrigin', 'anonymous');
   });
 
   it('renders js with SRI when present', () => {
-    const html = findRenderedDOMComponentWithTag(render(), 'html');
-    const js = html.querySelector('script[src]');
-    expect(js.getAttribute('integrity')).toEqual('sha512-disco-js');
-    expect(js.getAttribute('crossorigin')).toEqual('anonymous');
+    const root = render();
+    const js = root.find('script');
+
+    expect(js.at(1)).toHaveProp('integrity', 'sha512-disco-js');
+    expect(js.at(1)).toHaveProp('crossOrigin', 'anonymous');
   });
 
   it('renders state as JSON', () => {
-    const html = findRenderedDOMComponentWithTag(render(), 'html');
-    const json = html.querySelector('#redux-store-state').textContent;
-    expect(JSON.parse(json).foo).toEqual('bar');
+    const store = fakeStore;
+    const root = render({ store });
+    const json = root.find('#redux-store-state');
+
+    expect(json).toHaveLength(1);
+    expect(JSON.parse(json.prop('dangerouslySetInnerHTML').__html)).toEqual(
+      store.getState(),
+    );
   });
 
   it('renders meta with utf8 charset ', () => {
-    const html = findRenderedDOMComponentWithTag(render(), 'html');
-    const meta = html.querySelector('meta[charset]');
-    expect(meta.getAttribute('charset')).toEqual('utf-8');
+    const root = render();
+    const meta = root.find({ charSet: 'utf-8' });
+
+    expect(meta).toHaveLength(1);
   });
 
   it('renders favicon', () => {
     const amoCDN = 'https://test.cdn.net';
     const _config = getFakeConfig({ amoCDN });
-    const html = findRenderedDOMComponentWithTag(render({ _config }), 'html');
-    const favicon = html.querySelector('link[rel="shortcut icon"]');
-    expect(favicon.getAttribute('href')).toEqual(
+
+    const root = render({ _config });
+    const favicon = root.find('link[rel="shortcut icon"]');
+
+    expect(favicon).toHaveProp(
+      'href',
       `${amoCDN}/favicon.ico?v=${_config.get('faviconVersion')}`,
     );
   });
 
   it('renders title', () => {
-    const html = findRenderedDOMComponentWithTag(render(), 'html');
-    const title = html.querySelector('title').textContent;
-    expect(title).toEqual('test title');
+    const root = render();
+
+    expect(root.find('title')).toHaveText('test title');
```

Comment:
I know this is kind of how it was working before in the sense you could't override the title but is there a way to show that you could and that is the new title? Would that be useful here?

Reply:
I am not sure to know how to do it..